### PR TITLE
I10 unscented kalman filter

### PR DESCRIPTION
**Pull request by cbchau**
The new unscented kalman filter has refactored code into separate functions. In addition, the filter parameters have been tuned and the example made more similar to the assignment.

### CODE REVIEW

**Comment by yshiyi** [comment]
fig1
Please consider commenting on what each of state represents for and what the two inputs are.

**Comment by yshiyi** [comment]
fig2
I think "Q" defines the measurement disturbance. Maybe rephrase the comment.

**Comment by yshiyi** [comment]
fig3
Instead of listing the definitions of each input variable, how about put the definition and variable together as follows:
A: state matrix; B: input matrix; ... etc.

**Comment by yshiyi** [comment]
Overall, this is an excellent work. I have looked through your codes several times and don't think I have anything to suggest. Everything looks clean, neat and consistent.

**Comment by neginlashkari** [comment]
You have functionalized the code. Unfortunately, the effects of changes you have made to the previous codes was not clear to me. I think it is better to put more comments for your changes to show how did you improve the example, how you improved and tuned the filter. You can also define a parameter to show this new code is more optimized.

**Comment by cbchau** [comment]
1. Put comment on state and input to be position and velocity
2. Renamed the "Q" variable to motion disturbance
3. Updated all the functions to have more clear descriptions of variables in the comments
4. Example was changed to be more consistent with other examples
   4a. Input was separated out of the position matrix 'A' into a 'B' matrix
   4b. dt was taken out of the initial velocity at the beginning and applied during the position calculation in both EKF and UKF ex. x2 = A_x1 + B_v*dt
5. Filter was tuned such that the changing initial conditions,noise and variations in disturbances do not affect convergence
